File: transpiler/javatests/com/google/j2cl/transpiler/integration/allsimplebridges/Tester673.java
package com.google.j2cl.transpiler.integration.allsimplebridges;

import jsinterop.annotations.JsType;

public class Tester673 {
  @JsType
  static interface I1 {
    default String get(String value) {
      return "I1.get";
    }
  }

  @SuppressWarnings("unchecked")
  static class C1 implements I1 {
    @SuppressWarnings("MissingOverride")
    public String get(Object value) {
      return "C1.get";
    }
  }

  @SuppressWarnings("unchecked")
  public static void test() {
    C1 s = new C1();
    assert s.get(new Object()).equals("C1.get");
    assert ((I1) s).get("").equals("I1.get");
  }
}

File: transpiler/javatests/com/google/j2cl/integration/kotlin/allsimplebridges/Tester673.kt
/*
 * Copyright 2022 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package allsimplebridges

import com.google.j2cl.integration.testing.Asserts.assertTrue
import jsinterop.annotations.JsType

object Tester673 {
  @JsType
  internal interface I1 {
    fun get(value: String): String {
      return "I1.get"
    }
  }

  internal class C1 : I1 {
    fun get(value: Any): String {
      return "C1.get"
    }
  }

  fun test() {
    val s = C1()
    assertTrue(s.get(Any()) == "C1.get")
    assertTrue((s as I1).get("") == "I1.get")
  }
}
